### PR TITLE
change googleApiKey,

### PR DESCRIPTION
**Pull request by JustAnOtherCoder13**
add custom pinpoint on map,
change color of pinpoint when interested user on restaurant,
update restaurant list ui when user select a restaurant,
add position object to restaurant,
add number of interested colleague in param of restaurant model.

### CODE REVIEW

File: app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java [hunk 1]
```diff
@@ -127,4 +160,22 @@ private void updateLocationUI() {
             Log.e("Exception: %s", Objects.requireNonNull(e.getMessage()));
         }
     }
+
+
```
Comment (fodeguirassy) [inline app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java]: What if you do BitmapFactory.decodeResource(resources, R.drawable.id) ?

File: app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java [hunk 2]
```diff
@@ -71,8 +83,32 @@ public RestaurantViewModel(GetAllRestaurantsInteractor getAllRestaurantsInteract
 
     public LiveData<Restaurant> getSelectedRestaurant = selectedRestaurantMutableLiveData;
 
-    public List<Restaurant> getAllRestaurants() {
-        return getAllRestaurantsInteractor.getGeneratedRestaurants();
+    public LiveData<List<Restaurant>> getAllRestaurants = allRestaurantsMutableLiveData;
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @SuppressLint("CheckResult")
```
Comment (fodeguirassy) [inline app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java:93]: What is this for ?

File: app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java [hunk 3]
```diff
@@ -103,29 +148,27 @@ public void setUserToRestaurant() {
...
-    private void updateUserChosenRestaurant() {
+    public void updateUserChosenRestaurant() {
         User user = currentUserMutableLiveData.getValue();
         user.setUserDailySchedule(new UserDailySchedule(DATE, selectedRestaurantKeyMutableLiveData.getValue()));
         updateUserChosenRestaurantInteractor.updateUserChosenRestaurant(user)
                 .doFinally(() -> isDataLoadingMutableLiveData.setValue(false))
                 .andThen(getInterestedUsersForRestaurantKeyInteractor.getInterestedUsersForRestaurantKey(selectedRestaurantKeyMutableLiveData.getValue()))
-                .subscribe(users -> interestedUsersMutableLiveData.setValue(users)
+                .subscribe(users -> {
+                            persistedRestaurantMutableLiveData.getValue().setNumberOfInterestedUsers(users.size());
```
Comment (fodeguirassy) [inline app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java]: This is not Rx way. Chain observables.

File: app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java [hunk 4]
```diff
@@ -103,29 +148,27 @@ public void setUserToRestaurant() {
...
-    //Suppress warning is safe cause subscribe is only used to set _currentUser
-    @SuppressWarnings("ResultOfMethodCallIgnored")
-    @SuppressLint("CheckResult")
-    public void getCurrentUserForEmail(String authUserEmail) {
-        getCurrentUserForEmailInteractor.getCurrentUserForEmail(authUserEmail)
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(users -> currentUserMutableLiveData.setValue(users.get(0)));
+                    .subscribe(restaurantsForKey -> {
+                        int interestedUsers = restaurantsForKey.get(0).getNumberOfInterestedUsers() - 1;
```
Comment (fodeguirassy) [inline app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java:179]: This is not Rx way. Chain observables.

File: app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java (hunk 1 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java]: error :
D/skia: --- Failed to create image decoder with message 'unimplemented'
W/System.err: io.reactivex.exceptions.OnErrorNotImplementedException: Failed to decode image. The provided image must be a Bitmap.

File: app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java (hunk 3 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java]: done

File: app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java (hunk 4 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java:179]: done

File: app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java (hunk 2 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java:93]: cause i need an observeValueEvent on the restaurants list to update interested users in ui RestaurantList and MapsMarker

File: app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java [hunk 5]
```diff
@@ -127,4 +140,25 @@ private void updateLocationUI() {
             Log.e("Exception: %s", Objects.requireNonNull(e.getMessage()));
         }
     }
+
+    private void initCustomMarker() {
+        mRestaurantViewModel.getAllRestaurants.observe(getViewLifecycleOwner(), restaurants -> {
+            for (Restaurant restaurant : restaurants) {
+                LatLng restaurantLatLng = new LatLng
+                        (restaurant.getRestaurantPosition().getLatitude()
+                                , restaurant.getRestaurantPosition().getLongitude());
```
Comment (fodeguirassy) [inline app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java:149]: The icon is the only difference here. You may want to get the rest of the code out of the if block.

**Review by fodeguirassy — APPROVED** [review]

**Pull request merged**